### PR TITLE
Adding SpecLicense.html to javadoc classified artifacts on master

### PR DESCRIPTION
**Pull request by joakime**
+ Adding maven-javadoc-plugin configuration for:
  * description
  * doc title
  * window title
  * header
  * bottom (with links to mailing list and included EFSL license)

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

### CODE REVIEW

**Comment by joakime** [comment]
Note: I've opened Issue #306 about the missing `<organization>` and resulting bad Copyright line in the javadoc footer.
This PR is likely bad until that issue is answered / resolved.

File: api/pom.xml [hunk 1]
```diff
@@ -181,6 +181,15 @@
                     <links>
                         <link>http://docs.oracle.com/javase/6/docs/api/</link>
                     </links>
+                    <description>Jakarta WebSocket API documentation</description>
+                    <doctitle>Jakarta WebSocket API documentation</doctitle>
+                    <windowtitle>Jakarta WebSocket API documentation</windowtitle>
+                    <header><![CDATA[<br>Jakarta WebSocket API v${project.version}]]></header>
+                    <bottom><![CDATA[
+  Comments to: <a href="mailto:websocket-dev@eclipse.org">websocket-dev@eclipse.org</a>.<br>
+  Copyright &#169; 2017-{currentYear} - <a href="https://www.eclipse.org">Eclipse Foundation</a>. All rights reserved.<br>
```
Comment (m0mus) [inline api/pom.xml]: I think it should be 2012.

File: api/pom.xml [hunk 2]
```diff
@@ -171,7 +171,7 @@
                 <artifactId>maven-javadoc-plugin</artifactId>
                 <version>2.8.1</version>
```
Comment (m0mus) [inline api/pom.xml]: Does it make sense upgrading to version 3.1.0? You will be able to build on Java 11 in this case, but possibly will need to fix some javadocs.

File: api/pom.xml (hunk 1 above)
Comment (markt-asf) [reply api/pom.xml]: I agree with 2012 as the start date.

**Comment by joakime** [comment]
> Is there any way we can avoid have three exact copies of the spec license file in the repo? My concern is that they will get out of sync at some point.

Yes, but that would mean a new maven module/project and dependency that we could use with the maven-javadoc-plugin's `<resourcesArtifacts>` configuration.
http://maven.apache.org/plugins/maven-javadoc-plugin/jar-mojo.html#resourcesArtifacts

Probably a good idea for a higher level project in the Jakarta ecosystem to maintain that, and let all jakarta projects just use it.

File: api/pom.xml (hunk 1 above)
Comment (joakime) [reply api/pom.xml]: No thoughts on the fact that there's no Oracle copyright present here?

File: api/pom.xml (hunk 2 above)
Comment (joakime) [reply api/pom.xml]: Perhaps in the future.
I'd like to keep the changes to an absolute minimum in the short term.

File: api/pom.xml (hunk 1 above)
Comment (m0mus) [reply api/pom.xml]: Change the botton to this:
```
<bottom><![CDATA[
Copyright &#169; {inceptionYear}&#x2013;{currentYear} Oracle and/or its affiliates. All rights reserved.<br>
Use is subject to <a href="{@docRoot}/doc-files/EFSL.html" target="_top">license terms</a>.]]>
</bottom>
```
Override `inceptionYear' from the parent pom and set it's value to 2012.

It should fix everything.

File: api/pom.xml (hunk 2 above)
Comment (m0mus) [reply api/pom.xml]: It's not a blocking issue. I'm fine with leaving it as it is for now.

**Comment by m0mus** [comment]
Although I agree that it's nicer to have one copy of EFSL, it requires some efforts. The time is critical now (as usually) and I am suggesting doing it later (it you really want to do it).
It's unlikely that it went out of sync. I think we won't touch it in future.

File: api/pom.xml (hunk 1 above)
Comment (joakime) [reply api/pom.xml]: Fixed in commit 5599127ca22c67dd23078030f522b51524d4cfbe

**Pull request merged**